# Data analysis on the Dataset
This is a very general data analysis part on the dataset

In [1]:
%autosave 20
import pandas as pd

Autosaving every 20 seconds


### Load the data set
The file is in '*.txt', but the format of the data inside the file is in 'csv'. So load the data directly as 'csv' 

In [2]:
df=pd.read_csv("Sample dataset values.txt")

In [3]:
df

Sr No    Sensor                    Name Variable   Units  \
0    1586958692652188245  sensor03  PER_AIRMON_MESH1976150     PM10  ugm -3   
1    1586958692652348737  sensor01  PER_AIRMON_MESH1976150     PM 4  ugm -3   
2    1586958692652681524  sensor02  PER_AIRMON_MESH1976150       NO  ugm -3   
3    1586958692653617146  sensor01  PER_AIRMON_MESH1976150    PM2.5  ugm -3   
4    1586958692655668348  sensor02  PER_AIRMON_MESH1976150     PM 4  ugm -3   
..                   ...       ...                     ...      ...     ...   
262  1586958692890648280  sensor02  PER_AIRMON_MESH1976150     PM10  ugm -3   
263  1586958692890824556  sensor01  PER_AIRMON_MESH1976150     PM 4  ugm -3   
264  1586958692892191120  sensor03  PER_AIRMON_MESH1976150       O3     ppb   
265  1586958692892201991  sensor01  PER_AIRMON_MESH1976150       NO  ugm -3   
266  1586958692893048364  sensor02  PER_AIRMON_MESH1976150    PM2.5  ugm -3   

               Timestamp     Value  Flagged as Suspect Reading  
0    2018-12-13 03:16:00  16.12200                       False  
1    2018-12-14 12:46:00   5.19000                       False  
2    2018-12-14 00:30:00   0.08875                       False  
3    2018-12-14 12:46:00  13.84500                       False  
4    2018-12-14 00:30:00   1.76000                       False  
..                   ...       ...                         ...  
262  2018-12-14 00:42:00   7.38800                       False  
263  2018-12-14 13:03:00   4.21000                       False  
264  2018-12-13 03:20:00  23.01300                       False  
265  2018-12-14 13:04:00   5.38375                       False  
266  2018-12-14 00:42:00   3.51300                       False  

[267 rows x 8 columns]

See  that the first coulmn is the serial number and will not add anything to the analysis

In [14]:
df.columns[1:8,]

Index(['Sensor', 'Name', 'Variable', 'Units', 'Timestamp', 'Value',
       'Flagged as Suspect Reading'],
      dtype='object')

Index(['Sensor', 'Name', 'Variable', 'Units', 'Flagged as Suspect Reading'], dtype='object')

### Description of the objects and  not of the numerics

In [5]:
df.describe(include='O')

Sensor                    Name Variable   Units            Timestamp
count        267                     267      267     267                  267
unique         3                       1        8       3                   37
top     sensor01  PER_AIRMON_MESH1976150       NO  ugm -3  2018-12-14 13:00:00
freq         145                     267       36     200                    8

### Unique elements of the columns
The dataset has 8 columns and 267 records. Among these 'Sr No' , 'Timestamp' and 'Value' are numerics. Two of these fields are unique. 'Sr No' and 'Timestamp' are unique to each record. The column 'Values' are different for different sensors. The other columns there are repetitons.

In [6]:
def uni(col):
    print("There are ",len(df[col].unique())," unique elements  of column ", col," and they are ", df[col].unique())

In [7]:
list(map(uni,df[['Sensor', 'Name', 'Variable', 'Units','Flagged as Suspect Reading']]))

There are  3  unique elements  of column  Sensor  and they are  ['sensor03' 'sensor01' 'sensor02']
There are  1  unique elements  of column  Name  and they are  ['PER_AIRMON_MESH1976150']
There are  8  unique elements  of column  Variable  and they are  ['PM10' 'PM 4' 'NO' 'PM2.5' 'PM1' 'Particle Count' 'O3' 'NO2']
There are  3  unique elements  of column  Units  and they are  ['ugm -3' 'Kgm -3' 'ppb']
There are  2  unique elements  of column  Flagged as Suspect Reading  and they are  [False  True]


[None, None, None, None, None]